<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Treefit - Getting started • Treefit for R | Hayamizu lab</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Treefit - Getting started">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Treefit for R | Hayamizu lab</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/treefit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/working-with-seurat.html">Treefit - Working with Seurat</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/hayamizu-lab/treefit-r/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Treefit - Getting started</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/hayamizu-lab/treefit-r/blob/HEAD/vignettes/treefit.Rmd" class="external-link"><code>vignettes/treefit.Rmd</code></a></small>
      <div class="hidden name"><code>treefit.Rmd</code></div>

    </div>

    
    
<p>This document is a user-friendly manual on how to use the Treefit
package, which is the first toolkit for quantitative trajectory
inference using single-cell gene expression data. In this tutorial, we
demonstrate how to generate and analyze two kinds of toy data with the
aim to help get familiar with the practical workflow of Treefit. After
learning some basics, we will use Treefit to perform more biologically
interesting analysis in the next tutorial
(<code><a href="../articles/working-with-seurat.html">vignette("working-with-seurat")</a></code>).</p>
<div class="section level2">
<h2 id="generating-toy-data">1. Generating toy data<a class="anchor" aria-label="anchor" href="#generating-toy-data"></a>
</h2>
<p>While the Treefit package has been developed to help biologists who
wish to perform trajectory inference from single-cell gene expression
data, Treefit can also be viewed as a toolkit to generate and analyze a
point cloud in <i>d</i>-dimensional Euclidean space (<i>i.e.</i>,
simulated gene expression data).</p>
<p>Treefit provides some useful functions to generate artificial
datasets. For example, as we will now demonstrate, the function
<code><a href="../reference/generate_2d_n_arms_star_data.html">treefit::generate_2d_n_arms_star_data()</a></code> creates data that
approximately fit a star tree with the number of arms or branches; the
term star means a tree that looks like the star symbol “*”; for example,
the alphabet letters Y and X can be viewed as star trees that have three
and four arms, respectively.</p>
<p>The rows and columns of the generated data correspond to data points
(<i>i.e.</i>, <i>n</i> single cells) and their features (<i>i.e.</i>,
expression values of <i>d</i> different genes), respectively. The
Treefit package can be used to analyze both raw count data and
normalized expression data, but regarding the production of toy data, it
is meant to be used to generate continuous data like normalized
expression values.</p>
<p>Importantly, we can generate data with a desired level of noise by
changing the value of the <code>fatness</code> parameter of this
function. For example, if you set the <code>fatness</code> parameter to
<code>0.0</code> then you will get precisely tree-like data without
noise. By contrast, setting the <code>fatness</code> to <code>1.0</code>
gives very noisy data that are no longer tree-like. In this tutorial, we
deal with two types of toy data whose <code>fatness</code> values are
<code>0.1</code> and <code>0.8</code>, respectively. We note that
Treefit can be used to generate and analyze high dimensional datasets
but we focus on generating 2-dimensional data to make things as simple
as possible in this introductory tutorial.</p>
<div class="section level3">
<h3 id="tree-like-toy-data">1.1. Tree-like toy data<a class="anchor" aria-label="anchor" href="#tree-like-toy-data"></a>
</h3>
<p>Let us first generate 2-dimensional tree-like data that contain 500
data points and reasonably fit a star tree with three arms. We can
create such data and draw a scatter plot of them (Figure 1) simply by
executing the following two lines of code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">star.tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_2d_n_arms_star_data.html">generate_2d_n_arms_star_data</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">3</span>, <span class="fl">0.1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">star.tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/generate-tree-like-data-1.png" alt="Figure 1. A scatter plot of tree-like toy data generated by Treefit" width="700"><p class="caption">
Figure 1. A scatter plot of tree-like toy data generated by Treefit
</p>
</div>
</div>
<div class="section level3">
<h3 id="less-tree-like-noisy-toy-data">1.2. Less tree-like, noisy toy data<a class="anchor" aria-label="anchor" href="#less-tree-like-noisy-toy-data"></a>
</h3>
<p>Similarly, we can generate noisy data that do not look so tree-like
as the previous one. In this tutorial, we simply change the value of the
<code>fatness</code> parameter from <code>0.1</code> to <code>0.8</code>
in order to obtain non-tree-like data. The following two lines of code
yields a scatter plot shown in Figure 2:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">star.less_tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_2d_n_arms_star_data.html">generate_2d_n_arms_star_data</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">3</span>, <span class="fl">0.8</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">star.less_tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/generate-less-tree-like-data-1.png" alt="Figure 2. A scatter plot of noisy toy data generated by Treefit" width="700"><p class="caption">
Figure 2. A scatter plot of noisy toy data generated by Treefit
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="analyzing-the-data">2. Analyzing the data<a class="anchor" aria-label="anchor" href="#analyzing-the-data"></a>
</h2>
<p>Having generated two datasets whose tree-likeness are different, we
can analyze each of them. The Treefit package allows us to estimate how
well the data can be explained by tree models and to predict how many
“principal paths” there are in the best-fit tree. As shown in Figure 1,
the points in the first data clearly form a shape like the letter “Y”
and so the data are considered to fit a star tree with three arms very
well, whereas Figure 2 indicates that the second data are no longer very
tree-like because of the high level of added noise. Our goal in this
tutorial is to reproduce these conclusions by using Treefit.</p>
<div class="section level3">
<h3 id="tree-like-toy-data-1">2.1. Tree-like toy data<a class="anchor" aria-label="anchor" href="#tree-like-toy-data-1"></a>
</h3>
<p>Let us estimate the goodness-of-fit between tree models and the first
toy data. This can be done by using <code><a href="../reference/treefit.html">treefit::treefit()</a></code> as
follows. The <code>name</code> parameter is optional but we should
specify it, if possible. Because it’s useful to identify the
estimation.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit.tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/treefit.html">treefit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression<span class="op">=</span><span class="va">star.tree_like</span><span class="op">)</span>,
                                  name<span class="op">=</span><span class="st">"tree-like"</span><span class="op">)</span>
<span class="co"># Save the analytsis result to use other tutorials.</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">fit.tree_like</span>, <span class="st">"fit.tree_like.rds"</span><span class="op">)</span>
<span class="va">fit.tree_like</span></code></pre></div>
<pre><code><span class="co">## $name</span>
<span class="co">## [1] "tree-like"</span>
<span class="co">## </span>
<span class="co">## $max_cca_distance</span>
<span class="co">##     p       mean standard_deviation</span>
<span class="co">## 1   1 0.38870074        0.331060970</span>
<span class="co">## 2   2 0.05251131        0.010271991</span>
<span class="co">## 3   3 0.04927560        0.009498039</span>
<span class="co">## 4   4 0.04651486        0.008255324</span>
<span class="co">## 5   5 0.04454985        0.008529812</span>
<span class="co">## 6   6 0.04228183        0.007118898</span>
<span class="co">## 7   7 0.04030522        0.006579729</span>
<span class="co">## 8   8 0.03868173        0.006579729</span>
<span class="co">## 9   9 0.03764989        0.006172184</span>
<span class="co">## 10 10 0.03637713        0.005959552</span>
<span class="co">## 11 11 0.03508610        0.005552289</span>
<span class="co">## 12 12 0.03410683        0.005495294</span>
<span class="co">## 13 13 0.03378752        0.005611638</span>
<span class="co">## 14 14 0.03312892        0.005407782</span>
<span class="co">## 15 15 0.03212964        0.005497639</span>
<span class="co">## 16 16 0.03146133        0.005468667</span>
<span class="co">## 17 17 0.03055377        0.005626766</span>
<span class="co">## 18 18 0.02967156        0.005773466</span>
<span class="co">## 19 19 0.02894670        0.005324151</span>
<span class="co">## 20 20 0.02851185        0.005018886</span>
<span class="co">## </span>
<span class="co">## $rms_cca_distance</span>
<span class="co">##     p       mean standard_deviation</span>
<span class="co">## 1   1 0.38870074         0.33106097</span>
<span class="co">## 2   2 0.06331541         0.01070934</span>
<span class="co">## 3   3 0.09125438         0.01331089</span>
<span class="co">## 4   4 0.20143752         0.11178809</span>
<span class="co">## 5   5 0.13592156         0.01517657</span>
<span class="co">## 6   6 0.16331065         0.01983833</span>
<span class="co">## 7   7 0.25312806         0.06997458</span>
<span class="co">## 8   8 0.21430800         0.02400145</span>
<span class="co">## 9   9 0.24630525         0.02693568</span>
<span class="co">## 10 10 0.30090574         0.04956378</span>
<span class="co">## 11 11 0.30884577         0.03389472</span>
<span class="co">## 12 12 0.32303919         0.03475583</span>
<span class="co">## 13 13 0.38793748         0.02618646</span>
<span class="co">## 14 14 0.38838038         0.02965039</span>
<span class="co">## 15 15 0.38690295         0.02908631</span>
<span class="co">## 16 16 0.36878069         0.03155892</span>
<span class="co">## 17 17 0.38431219         0.03031101</span>
<span class="co">## 18 18 0.41102470         0.02574436</span>
<span class="co">## 19 19 0.43575051         0.02502915</span>
<span class="co">## 20 20 0.44683230         0.02192529</span>
<span class="co">## </span>
<span class="co">## $n_principal_paths_candidates</span>
<span class="co">## [1]  3  6  9 17</span>
<span class="co">## </span>
<span class="co">## attr(,"class")</span>
<span class="co">## [1] "treefit"</span></code></pre>
<p><code><a href="../reference/treefit.html">treefit::treefit()</a></code> returns a <code>treefit</code> object
that summarizes the analysis of Treefit. We will explain how to
interpret the results in the next section. For now, we may focus on
learning how to use Treefit.</p>
<p>As we will see later, it is helpful to visualize the results using
<code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code>. By executing <code>plot(fit.tree_like)</code>, we
can obtain the following two user-friendly visual plots, which makes it
easier to interpret the results of the Treefit analysis.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit.tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/plot-tree-like-data-estimation-result-1.png" alt="Figure 3. The output for the tree-like data shown in Figure 1" width="700"><p class="caption">
Figure 3. The output for the tree-like data shown in Figure 1
</p>
</div>
</div>
<div class="section level3">
<h3 id="less-tree-like-noisy-toy-data-1">2.2. Less tree-like, noisy toy data<a class="anchor" aria-label="anchor" href="#less-tree-like-noisy-toy-data-1"></a>
</h3>
<p>We can analyze the second toy data in the same manner.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit.less_tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/treefit.html">treefit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression<span class="op">=</span><span class="va">star.less_tree_like</span><span class="op">)</span>,
                                       name<span class="op">=</span><span class="st">"less-tree-like"</span><span class="op">)</span>
<span class="co"># Save the analytsis result to use other tutorials.</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">fit.less_tree_like</span>, <span class="st">"fit.less_tree_like.rds"</span><span class="op">)</span>
<span class="va">fit.less_tree_like</span></code></pre></div>
<pre><code><span class="co">## $name</span>
<span class="co">## [1] "less-tree-like"</span>
<span class="co">## </span>
<span class="co">## $max_cca_distance</span>
<span class="co">##     p       mean standard_deviation</span>
<span class="co">## 1   1 0.66677748         0.18273424</span>
<span class="co">## 2   2 0.35061875         0.08848348</span>
<span class="co">## 3   3 0.28896555         0.06994095</span>
<span class="co">## 4   4 0.23027723         0.04277271</span>
<span class="co">## 5   5 0.21744564         0.04340571</span>
<span class="co">## 6   6 0.20426248         0.04335538</span>
<span class="co">## 7   7 0.17215283         0.03212272</span>
<span class="co">## 8   8 0.15531972         0.02654320</span>
<span class="co">## 9   9 0.14395646         0.02808796</span>
<span class="co">## 10 10 0.13052459         0.02270056</span>
<span class="co">## 11 11 0.12801079         0.02143530</span>
<span class="co">## 12 12 0.12133702         0.01768292</span>
<span class="co">## 13 13 0.11750223         0.01936427</span>
<span class="co">## 14 14 0.11019091         0.02058131</span>
<span class="co">## 15 15 0.10482574         0.02066918</span>
<span class="co">## 16 16 0.09941680         0.02011737</span>
<span class="co">## 17 17 0.09518913         0.02025271</span>
<span class="co">## 18 18 0.08834924         0.01868935</span>
<span class="co">## 19 19 0.08328521         0.01671675</span>
<span class="co">## 20 20 0.07905554         0.01589106</span>
<span class="co">## </span>
<span class="co">## $rms_cca_distance</span>
<span class="co">##     p      mean standard_deviation</span>
<span class="co">## 1   1 0.6667775         0.18273424</span>
<span class="co">## 2   2 0.4791963         0.10732119</span>
<span class="co">## 3   3 0.5316548         0.13158219</span>
<span class="co">## 4   4 0.5264191         0.09747992</span>
<span class="co">## 5   5 0.4973448         0.08857052</span>
<span class="co">## 6   6 0.5102476         0.07048500</span>
<span class="co">## 7   7 0.5350198         0.04880854</span>
<span class="co">## 8   8 0.5393634         0.05219577</span>
<span class="co">## 9   9 0.5515511         0.04468544</span>
<span class="co">## 10 10 0.5564527         0.03733736</span>
<span class="co">## 11 11 0.5659675         0.03830011</span>
<span class="co">## 12 12 0.5692597         0.03989553</span>
<span class="co">## 13 13 0.5826631         0.03382459</span>
<span class="co">## 14 14 0.5893211         0.03430985</span>
<span class="co">## 15 15 0.5824711         0.03221168</span>
<span class="co">## 16 16 0.5855871         0.03016659</span>
<span class="co">## 17 17 0.5842967         0.02911714</span>
<span class="co">## 18 18 0.5838863         0.03015288</span>
<span class="co">## 19 19 0.5847512         0.02969265</span>
<span class="co">## 20 20 0.5841810         0.03730001</span>
<span class="co">## </span>
<span class="co">## $n_principal_paths_candidates</span>
<span class="co">## [1]  3  6 16 19</span>
<span class="co">## </span>
<span class="co">## attr(,"class")</span>
<span class="co">## [1] "treefit"</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit.less_tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/plot-less-tree-like-data-estimation-result-1.png" alt="Figure 4. The output for the noisy data shown in Figure 2" width="700"><p class="caption">
Figure 4. The output for the noisy data shown in Figure 2
</p>
</div>
</div>
<div class="section level3">
<h3 id="comparing-two-results">2.3. Comparing two results<a class="anchor" aria-label="anchor" href="#comparing-two-results"></a>
</h3>
<p>We can compare different results by passing all results to
<code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> as follows:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit.tree_like</span>, <span class="va">fit.less_tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/plot-estimation-results-1.png" alt="Figure 5. Comparison of the plots shown in Figure 3 and Figure 4." width="700"><p class="caption">
Figure 5. Comparison of the plots shown in Figure 3 and Figure 4.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="interpreting-the-results">3. Interpreting the results<a class="anchor" aria-label="anchor" href="#interpreting-the-results"></a>
</h2>
<p>Before interpreting the previous results, we briefly summarize the
process of the Treefit analysis that consists of the following three
steps.</p>
<ol style="list-style-type: decimal">
<li><p>First, Treefit repeatedly “perturbs” the input data (<i>i.e.</i>,
adds some small noise to the original row count data or normalized
expression data) in order to produce many slightly different datasets
that may have been acquired in the biological experiment.</p></li>
<li><p>Second, for each dataset, Treefit calculates a distance matrix
that represents the dissimilarities between sample cells and then
constructs a tree from each distance matrix. The current version of
Treefit computes a minimum spanning tree (MST) that has been widely used
for trajectory inference.</p></li>
<li><p>Finally, Treefit evaluates the goodness-of-fit between the data
and tree models. The underlying idea of this method is that the
structure of trees inferred from tree-like data tends to have high
robustness to noise, compared to non-tree-like data. Therefore, Treefit
measures the mutual similarity between estimated trees in order to check
the stability of the tree structures. To this end, Treefit constructs a
<i>p</i>-dimensional subspace that extracts the main features of each
tree structure and then measuring mutual similarities between the
subspaces by using a special type of metrics called the Grassmann
distance. In principle, when the estimated trees are mutually similar in
their structure, the mean and standard deviation (SD) of the Grassmann
distance are small.</p></li>
</ol>
<p>Although the word “Grassmann distance” may sound so unfamiliar to
some readers, the concept appears in different disguises in various
practical contexts. For example, the Grassmann distance has a close
connection to canonical correlation analysis (CCA). Treefit provides two
Grassmann distances <code>$max_cca_distance</code> and
<code>$rms_cca_distance</code> that can be used for different purposes
as we now explain.</p>
<div class="section level3">
<h3 id="estimating-the-tree-likeness-of-data">3.1. Estimating the tree-likeness of data<a class="anchor" aria-label="anchor" href="#estimating-the-tree-likeness-of-data"></a>
</h3>
<p>The Treefit analysis using the first Grassmann distance
<code>$max_cca_distance</code> (shown in the left panel of Figure 5)
tells us the goodness-of-fit between data and tree models. In principle,
as mentioned earlier, if the mean and SD of
<code>$max_cca_distance</code> are small, then this means that the
estimated trees are mutually similar in their structure. As can be
observed, the distance changes according to the dimensionality <i>p</i>
of the feature space, but <code>$max_cca_distance</code> has the
property that the value decreases monotonically as <i>p</i> increases
for any datasets.</p>
<p>Comparing the Treefit results for the two datasets, we see that the
mean Grassmann distance for the first data does not fall below the
second one regardless of the value of <i>p</i> and that the SD of the
Grassmann distance for the first data is very small compared to the
second data. These results imply that the estimated tree structures are
very robust to noise in the first case but not in the second case. Thus,
Treefit has verified that the first data are highly tree-like while the
second data are not.</p>
</div>
<div class="section level3">
<h3 id="predicting-the-number-of-principal-paths-in-the-underlying-tree">3.2. Predicting the number of principal paths in the underlying
tree<a class="anchor" aria-label="anchor" href="#predicting-the-number-of-principal-paths-in-the-underlying-tree"></a>
</h3>
<p>The Treefit analysis using the other Grassmann distance
<code>$rms_cca_distance</code> (shown in the right panel of Figure 5) is
useful to infer the number of “principal paths” in the best-fit tree.
From a biological perspective, this analysis can be used to discover a
novel or unexpected cell type from single-cell gene expression (for
details, see <code><a href="../articles/working-with-seurat.html">vignette("working-with-seurat")</a></code>).</p>
<p>Unlike the previous Grassmann distance, the mean value of
<code>$rms_cca_distance</code> can fluctuate depending on the value of
<i>p</i>. Interestingly, we can predict the number of principal paths in
the best-fit tree by exploring for which <i>p</i> the distance value
reaches “the bottom of a valley” (<i>i.e.</i>, attains a local minimum).
More precisely, when <code>$rms_cca_distance</code> attains a local
minimum at a certain <i>p</i>, the value <i>p</i>+1 indicates the number
of principal paths in the best-fit tree.
<code>$n_principal_paths_candidates</code> has these <i>p</i>+1 values.
We don’t need to calculate them manually. When Treefit produces a plot
having more than one valleys, the smallest <i>p</i> is usually most
informative for the prediction. The smallest <i>p</i>+1 value can be
obtained by <code>$n_principal_paths_candidates[1]</code>.</p>
<p>Comparing the Treefit results for the two datasets, we first see that
both plots attains a local minimum at <i>p</i>=2. This means that for
both datasets the best-fit tree has <i>p</i>+1=3 principle paths, which
is correct because both were generated from the same star tree with
three arms. Another important point to be made is that the SD of the
Grassmann distance for the first data is very small at <i>p</i>=2
compared to that for the second data; in other words, Treefit made this
prediction more confidently for the first dataset than for the second
one. This result is reasonable because the first dataset is much less
noisy than the second one. Thus, Treefit has correctly determined the
number of principal paths in the underlying tree together with the
goodness-of-fit for each dataset.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Momoko Hayamizu, Kouhei Sutou, Ryohei Suzuki, Hiromi Ishii.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
