<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Treefit - Getting started • Treefit for R | Hayamizu lab</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Treefit - Getting started">
<meta property="og:description" content="treefit">
<meta property="og:image" content="https://hayamizu-lab.github.io/treefit-r/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Treefit for R | Hayamizu lab</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/treefit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/working-with-seurat.html">Treefit - Working with Seurat</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/hayamizu-lab/treefit-r/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="treefit_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Treefit - Getting started</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/hayamizu-lab/treefit-r/blob/HEAD/vignettes/treefit.Rmd" class="external-link"><code>vignettes/treefit.Rmd</code></a></small>
      <div class="hidden name"><code>treefit.Rmd</code></div>

    </div>

    
    
<p>This document is a user-friendly manual on how to use the Treefit package, which is the first toolkit for quantitative trajectory inference using single-cell gene expression data. In this tutorial, we demonstrate how to generate and analyze two kinds of toy data with the aim to help get familiar with the practical workflow of Treefit. After learning some basics, we will use Treefit to perform more biologically interesting analysis in the next tutorial (<code><a href="../articles/working-with-seurat.html">vignette("working-with-seurat")</a></code>).</p>
<div class="section level2">
<h2 id="generating-toy-data">1. Generating toy data<a class="anchor" aria-label="anchor" href="#generating-toy-data"></a>
</h2>
<p>While the Treefit package has been developed to help biologists who wish to perform trajectory inference from single-cell gene expression data, Treefit can also be viewed as a toolkit to generate and analyze a point cloud in <i>d</i>-dimensional Euclidean space (<i>i.e.</i>, simulated gene expression data).</p>
<p>Treefit provides some useful functions to generate artificial datasets. For example, as we will now demonstrate, the function <code><a href="../reference/generate_2d_n_arms_star_data.html">treefit::generate_2d_n_arms_star_data()</a></code> creates data that approximately fit a star tree with the number of arms or branches; the term star means a tree that looks like the star symbol “*”; for example, the alphabet letters Y and X can be viewed as star trees that have three and four arms, respectively.</p>
<p>The rows and columns of the generated data correspond to data points (<i>i.e.</i>, <i>n</i> single cells) and their features (<i>i.e.</i>, expression values of <i>d</i> different genes), respectively. The Treefit package can be used to analyze both raw count data and normalized expression data, but regarding the production of toy data, it is meant to be used to generate continuous data like normalized expression values.</p>
<p>Importantly, we can generate data with a desired level of noise by changing the value of the <code>fatness</code> parameter of this function. For example, if you set the <code>fatness</code> parameter to <code>0.0</code> then you will get precisely tree-like data without noise. By contrast, setting the <code>fatness</code> to <code>1.0</code> gives very noisy data that are no longer tree-like. In this tutorial, we deal with two types of toy data whose <code>fatness</code> values are <code>0.1</code> and <code>0.8</code>, respectively. We note that Treefit can be used to generate and analyze high dimensional datasets but we focus on generating 2-dimensional data to make things as simple as possible in this introductory tutorial.</p>
<div class="section level3">
<h3 id="tree-like-toy-data">1.1. Tree-like toy data<a class="anchor" aria-label="anchor" href="#tree-like-toy-data"></a>
</h3>
<p>Let us first generate 2-dimensional tree-like data that contain 500 data points and reasonably fit a star tree with three arms. We can create such data and draw a scatter plot of them (Figure 1) simply by executing the following two lines of code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">star.tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_2d_n_arms_star_data.html">generate_2d_n_arms_star_data</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">3</span>, <span class="fl">0.1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">star.tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/generate-tree-like-data-1.png" alt="Figure 1. A scatter plot of tree-like toy data generated by Treefit" width="700"><p class="caption">
Figure 1. A scatter plot of tree-like toy data generated by Treefit
</p>
</div>
</div>
<div class="section level3">
<h3 id="less-tree-like-noisy-toy-data">1.2. Less tree-like, noisy toy data<a class="anchor" aria-label="anchor" href="#less-tree-like-noisy-toy-data"></a>
</h3>
<p>Similarly, we can generate noisy data that do not look so tree-like as the previous one. In this tutorial, we simply change the value of the <code>fatness</code> parameter from <code>0.1</code> to <code>0.8</code> in order to obtain non-tree-like data. The following two lines of code yields a scatter plot shown in Figure 2:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">star.less_tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/generate_2d_n_arms_star_data.html">generate_2d_n_arms_star_data</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">3</span>, <span class="fl">0.8</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">star.less_tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/generate-less-tree-like-data-1.png" alt="Figure 2. A scatter plot of noisy toy data generated by Treefit" width="700"><p class="caption">
Figure 2. A scatter plot of noisy toy data generated by Treefit
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="analyzing-the-data">2. Analyzing the data<a class="anchor" aria-label="anchor" href="#analyzing-the-data"></a>
</h2>
<p>Having generated two datasets whose tree-likeness are different, we can analyze each of them. The Treefit package allows us to estimate how well the data can be explained by tree models and to predict how many “principal paths” there are in the best-fit tree. As shown in Figure 1, the points in the first data clearly form a shape like the letter “Y” and so the data are considered to fit a star tree with three arms very well, whereas Figure 2 indicates that the second data are no longer very tree-like because of the high level of added noise. Our goal in this tutorial is to reproduce these conclusions by using Treefit.</p>
<div class="section level3">
<h3 id="tree-like-toy-data-1">2.1. Tree-like toy data<a class="anchor" aria-label="anchor" href="#tree-like-toy-data-1"></a>
</h3>
<p>Let us estimate the goodness-of-fit between tree models and the first toy data. This can be done by using <code><a href="../reference/treefit.html">treefit::treefit()</a></code> as follows. The <code>name</code> parameter is optional but we should specify it, if possible. Because it’s useful to identify the estimation.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit.tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/treefit.html">treefit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression<span class="op">=</span><span class="va">star.tree_like</span><span class="op">)</span>,
                                  name<span class="op">=</span><span class="st">"tree-like"</span><span class="op">)</span>
<span class="co"># Save the analytsis result to use other tutorials.</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">fit.tree_like</span>, <span class="st">"fit.tree_like.rds"</span><span class="op">)</span>
<span class="va">fit.tree_like</span></code></pre></div>
<pre><code><span class="co">## $name</span>
<span class="co">## [1] "tree-like"</span>
<span class="co">## </span>
<span class="co">## $max_cca_distance</span>
<span class="co">##     p       mean standard_deviation</span>
<span class="co">## 1   1 0.18860598        0.112087294</span>
<span class="co">## 2   2 0.05734910        0.006932817</span>
<span class="co">## 3   3 0.05028274        0.006170543</span>
<span class="co">## 4   4 0.04734358        0.006528380</span>
<span class="co">## 5   5 0.04570173        0.006737911</span>
<span class="co">## 6   6 0.04366379        0.006193305</span>
<span class="co">## 7   7 0.04003216        0.006509366</span>
<span class="co">## 8   8 0.03889504        0.006438062</span>
<span class="co">## 9   9 0.03777213        0.006137765</span>
<span class="co">## 10 10 0.03642294        0.006309629</span>
<span class="co">## 11 11 0.03568197        0.006243237</span>
<span class="co">## 12 12 0.03442762        0.006171836</span>
<span class="co">## 13 13 0.03354193        0.006166322</span>
<span class="co">## 14 14 0.03247914        0.006430830</span>
<span class="co">## 15 15 0.03156625        0.005902780</span>
<span class="co">## 16 16 0.03089184        0.006015070</span>
<span class="co">## 17 17 0.03042155        0.006119202</span>
<span class="co">## 18 18 0.03001159        0.006023056</span>
<span class="co">## 19 19 0.02884276        0.005819269</span>
<span class="co">## 20 20 0.02790484        0.005380303</span>
<span class="co">## </span>
<span class="co">## $rms_cca_distance</span>
<span class="co">##     p       mean standard_deviation</span>
<span class="co">## 1   1 0.18860598         0.11208729</span>
<span class="co">## 2   2 0.07706869         0.01352671</span>
<span class="co">## 3   3 0.10971008         0.02013602</span>
<span class="co">## 4   4 0.17194455         0.04471201</span>
<span class="co">## 5   5 0.16888451         0.01674156</span>
<span class="co">## 6   6 0.18068421         0.01660333</span>
<span class="co">## 7   7 0.20414822         0.02973250</span>
<span class="co">## 8   8 0.21943624         0.02177501</span>
<span class="co">## 9   9 0.26109335         0.02527100</span>
<span class="co">## 10 10 0.30830288         0.03879098</span>
<span class="co">## 11 11 0.32295636         0.02381525</span>
<span class="co">## 12 12 0.33768563         0.02718667</span>
<span class="co">## 13 13 0.35599976         0.02567791</span>
<span class="co">## 14 14 0.37155751         0.03141696</span>
<span class="co">## 15 15 0.38831341         0.02749337</span>
<span class="co">## 16 16 0.40683549         0.02607011</span>
<span class="co">## 17 17 0.43104143         0.02428271</span>
<span class="co">## 18 18 0.43955155         0.02075421</span>
<span class="co">## 19 19 0.45054377         0.02611018</span>
<span class="co">## 20 20 0.45909418         0.02687859</span>
<span class="co">## </span>
<span class="co">## $n_principal_paths_candidates</span>
<span class="co">## [1] 3 6</span>
<span class="co">## </span>
<span class="co">## attr(,"class")</span>
<span class="co">## [1] "treefit"</span></code></pre>
<p><code><a href="../reference/treefit.html">treefit::treefit()</a></code> returns a <code>treefit</code> object that summarizes the analysis of Treefit. We will explain how to interpret the results in the next section. For now, we may focus on learning how to use Treefit.</p>
<p>As we will see later, it is helpful to visualize the results using <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code>. By executing <code>plot(fit.tree_like)</code>, we can obtain the following two user-friendly visual plots, which makes it easier to interpret the results of the Treefit analysis.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit.tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/plot-tree-like-data-estimation-result-1.png" alt="Figure 3. The output for the tree-like data shown in Figure 1" width="700"><p class="caption">
Figure 3. The output for the tree-like data shown in Figure 1
</p>
</div>
</div>
<div class="section level3">
<h3 id="less-tree-like-noisy-toy-data-1">2.2. Less tree-like, noisy toy data<a class="anchor" aria-label="anchor" href="#less-tree-like-noisy-toy-data-1"></a>
</h3>
<p>We can analyze the second toy data in the same manner.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit.less_tree_like</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/treefit.html">treefit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression<span class="op">=</span><span class="va">star.less_tree_like</span><span class="op">)</span>,
                                       name<span class="op">=</span><span class="st">"less-tree-like"</span><span class="op">)</span>
<span class="co"># Save the analytsis result to use other tutorials.</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">fit.less_tree_like</span>, <span class="st">"fit.less_tree_like.rds"</span><span class="op">)</span>
<span class="va">fit.less_tree_like</span></code></pre></div>
<pre><code><span class="co">## $name</span>
<span class="co">## [1] "less-tree-like"</span>
<span class="co">## </span>
<span class="co">## $max_cca_distance</span>
<span class="co">##     p      mean standard_deviation</span>
<span class="co">## 1   1 0.6410779         0.16530427</span>
<span class="co">## 2   2 0.3503794         0.05149123</span>
<span class="co">## 3   3 0.2612743         0.04514520</span>
<span class="co">## 4   4 0.2401434         0.04251065</span>
<span class="co">## 5   5 0.2166526         0.03032552</span>
<span class="co">## 6   6 0.2067030         0.03195008</span>
<span class="co">## 7   7 0.1978107         0.03042195</span>
<span class="co">## 8   8 0.1896353         0.02809781</span>
<span class="co">## 9   9 0.1722665         0.02535623</span>
<span class="co">## 10 10 0.1594339         0.02559466</span>
<span class="co">## 11 11 0.1506703         0.02653699</span>
<span class="co">## 12 12 0.1465941         0.02491429</span>
<span class="co">## 13 13 0.1388525         0.02562315</span>
<span class="co">## 14 14 0.1298897         0.02500551</span>
<span class="co">## 15 15 0.1244570         0.02428170</span>
<span class="co">## 16 16 0.1202338         0.02334609</span>
<span class="co">## 17 17 0.1135588         0.02206775</span>
<span class="co">## 18 18 0.1107619         0.02161223</span>
<span class="co">## 19 19 0.1058824         0.02154103</span>
<span class="co">## 20 20 0.1010044         0.02193400</span>
<span class="co">## </span>
<span class="co">## $rms_cca_distance</span>
<span class="co">##     p      mean standard_deviation</span>
<span class="co">## 1   1 0.6410779         0.16530427</span>
<span class="co">## 2   2 0.3952325         0.04530719</span>
<span class="co">## 3   3 0.5374504         0.06735476</span>
<span class="co">## 4   4 0.6169260         0.04663143</span>
<span class="co">## 5   5 0.6171774         0.03651230</span>
<span class="co">## 6   6 0.6132443         0.03881056</span>
<span class="co">## 7   7 0.6113552         0.03363042</span>
<span class="co">## 8   8 0.6265815         0.02904500</span>
<span class="co">## 9   9 0.6332530         0.02639997</span>
<span class="co">## 10 10 0.6326131         0.02172755</span>
<span class="co">## 11 11 0.6198775         0.02909744</span>
<span class="co">## 12 12 0.6009167         0.02860706</span>
<span class="co">## 13 13 0.5991288         0.02447413</span>
<span class="co">## 14 14 0.5998079         0.02713869</span>
<span class="co">## 15 15 0.6018131         0.02720529</span>
<span class="co">## 16 16 0.6019081         0.03177225</span>
<span class="co">## 17 17 0.6023813         0.03527602</span>
<span class="co">## 18 18 0.5943385         0.03445902</span>
<span class="co">## 19 19 0.5863388         0.03584081</span>
<span class="co">## 20 20 0.5764867         0.04072415</span>
<span class="co">## </span>
<span class="co">## $n_principal_paths_candidates</span>
<span class="co">## [1]  3  8 14</span>
<span class="co">## </span>
<span class="co">## attr(,"class")</span>
<span class="co">## [1] "treefit"</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit.less_tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/plot-less-tree-like-data-estimation-result-1.png" alt="Figure 4. The output for the noisy data shown in Figure 2" width="700"><p class="caption">
Figure 4. The output for the noisy data shown in Figure 2
</p>
</div>
</div>
<div class="section level3">
<h3 id="comparing-two-results">2.3. Comparing two results<a class="anchor" aria-label="anchor" href="#comparing-two-results"></a>
</h3>
<p>We can compare different results by passing all results to <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> as follows:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit.tree_like</span>, <span class="va">fit.less_tree_like</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="treefit_files/figure-html/plot-estimation-results-1.png" alt="Figure 5. Comparison of the plots shown in Figure 3 and Figure 4." width="700"><p class="caption">
Figure 5. Comparison of the plots shown in Figure 3 and Figure 4.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="interpreting-the-results">3. Interpreting the results<a class="anchor" aria-label="anchor" href="#interpreting-the-results"></a>
</h2>
<p>Before interpreting the previous results, we briefly summarize the process of the Treefit analysis that consists of the following three steps.</p>
<ol style="list-style-type: decimal">
<li><p>First, Treefit repeatedly “perturbs” the input data (<i>i.e.</i>, adds some small noise to the original row count data or normalized expression data) in order to produce many slightly different datasets that may have been acquired in the biological experiment.</p></li>
<li><p>Second, for each dataset, Treefit calculates a distance matrix that represents the dissimilarities between sample cells and then constructs a tree from each distance matrix. The current version of Treefit computes a minimum spanning tree (MST) that has been widely used for trajectory inference.</p></li>
<li><p>Finally, Treefit evaluates the goodness-of-fit between the data and tree models. The underlying idea of this method is that the structure of trees inferred from tree-like data tends to have high robustness to noise, compared to non-tree-like data. Therefore, Treefit measures the mutual similarity between estimated trees in order to check the stability of the tree structures. To this end, Treefit constructs a <i>p</i>-dimensional subspace that extracts the main features of each tree structure and then measuring mutual similarities between the subspaces by using a special type of metrics called the Grassmann distance. In principle, when the estimated trees are mutually similar in their structure, the mean and standard deviation (SD) of the Grassmann distance are small.</p></li>
</ol>
<p>Although the word “Grassmann distance” may sound so unfamiliar to some readers, the concept appears in different disguises in various practical contexts. For example, the Grassmann distance has a close connection to canonical correlation analysis (CCA). Treefit provides two Grassmann distances <code>$max_cca_distance</code> and <code>$rms_cca_distance</code> that can be used for different purposes as we now explain.</p>
<div class="section level3">
<h3 id="estimating-the-tree-likeness-of-data">3.1. Estimating the tree-likeness of data<a class="anchor" aria-label="anchor" href="#estimating-the-tree-likeness-of-data"></a>
</h3>
<p>The Treefit analysis using the first Grassmann distance <code>$max_cca_distance</code> (shown in the left panel of Figure 5) tells us the goodness-of-fit between data and tree models. In principle, as mentioned earlier, if the mean and SD of <code>$max_cca_distance</code> are small, then this means that the estimated trees are mutually similar in their structure. As can be observed, the distance changes according to the dimensionality <i>p</i> of the feature space, but <code>$max_cca_distance</code> has the property that the value decreases monotonically as <i>p</i> increases for any datasets.</p>
<p>Comparing the Treefit results for the two datasets, we see that the mean Grassmann distance for the first data does not fall below the second one regardless of the value of <i>p</i> and that the SD of the Grassmann distance for the first data is very small compared to the second data. These results imply that the estimated tree structures are very robust to noise in the first case but not in the second case. Thus, Treefit has verified that the first data are highly tree-like while the second data are not.</p>
</div>
<div class="section level3">
<h3 id="predicting-the-number-of-principal-paths-in-the-underlying-tree">3.2. Predicting the number of principal paths in the underlying tree<a class="anchor" aria-label="anchor" href="#predicting-the-number-of-principal-paths-in-the-underlying-tree"></a>
</h3>
<p>The Treefit analysis using the other Grassmann distance <code>$rms_cca_distance</code> (shown in the right panel of Figure 5) is useful to infer the number of “principal paths” in the best-fit tree. From a biological perspective, this analysis can be used to discover a novel or unexpected cell type from single-cell gene expression (for details, see <code><a href="../articles/working-with-seurat.html">vignette("working-with-seurat")</a></code>).</p>
<p>Unlike the previous Grassmann distance, the mean value of <code>$rms_cca_distance</code> can fluctuate depending on the value of <i>p</i>. Interestingly, we can predict the number of principal paths in the best-fit tree by exploring for which <i>p</i> the distance value reaches “the bottom of a valley” (<i>i.e.</i>, attains a local minimum). More precisely, when <code>$rms_cca_distance</code> attains a local minimum at a certain <i>p</i>, the value <i>p</i>+1 indicates the number of principal paths in the best-fit tree. <code>$n_principal_paths_candidates</code> has these <i>p</i>+1 values. We don’t need to calculate them manually. When Treefit produces a plot having more than one valleys, the smallest <i>p</i> is usually most informative for the prediction. The smallest <i>p</i>+1 value can be obtained by <code>$n_principal_paths_candidates[1]</code>.</p>
<p>Comparing the Treefit results for the two datasets, we first see that both plots attains a local minimum at <i>p</i>=2. This means that for both datasets the best-fit tree has <i>p</i>+1=3 principle paths, which is correct because both were generated from the same star tree with three arms. Another important point to be made is that the SD of the Grassmann distance for the first data is very small at <i>p</i>=2 compared to that for the second data; in other words, Treefit made this prediction more confidently for the first dataset than for the second one. This result is reasonable because the first dataset is much less noisy than the second one. Thus, Treefit has correctly determined the number of principal paths in the underlying tree together with the goodness-of-fit for each dataset.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Momoko Hayamizu, Kouhei Sutou, Ryohei Suzuki, Hiromi Ishii.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
