<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Treefit - Working with Seurat • Treefit for R | Hayamizu lab</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Treefit - Working with Seurat">
<meta property="og:description" content="treefit">
<meta property="og:image" content="https://hayamizu-lab.github.io/treefit-r/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Treefit for R | Hayamizu lab</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/treefit.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/working-with-seurat.html">Treefit - Working with Seurat</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/hayamizu-lab/treefit-r/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="working-with-seurat_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Treefit - Working with Seurat</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/hayamizu-lab/treefit-r/blob/HEAD/vignettes/working-with-seurat.Rmd" class="external-link"><code>vignettes/working-with-seurat.Rmd</code></a></small>
      <div class="hidden name"><code>working-with-seurat.Rmd</code></div>

    </div>

    
    
<p>This is a user-friendly manual for biologists who are interested in performing quantitative trajectory inference with the Treefit package. If you are not familiar with the basic workflow of Treefit, then it would be a good idea to look at the introductory tutorial (<code><a href="../articles/treefit.html">vignette("treefit")</a></code>) before using it.</p>
<div class="section level2">
<h2 id="why-quantitative-trajectory-inference">1. Why “quantitative” trajectory inference?<a class="anchor" aria-label="anchor" href="#why-quantitative-trajectory-inference"></a>
</h2>
<p>Single-cell technologies are expected to help us discover a novel type of cells and revolutionize our understanding of the process of cell differentiation, and trajectory inference is one of the key computational challenges in single-cell transcriptomics. In recent years, many software packages have been developed and widely used to extract an underlying “tree” trajectory from single-cell RNA-seq data.</p>
<p>However, trajectory inference often suffers from the uncertainty due to the heterogeneity of individual cells and the high levels of technical noise in single-cell experiments. Hence, it was a fatal problem that there was no method to quantitatively assess the reliability of the estimated trajectory and to distinguish distinct cell types in an objective manner so far. Of course, many handy software packages are available for visualizing data and helping perform exploratory data analysis; however, we must stress that different visualization techniques often result in completely different pictures and the interpretation can be quite subjective and thus it is hard to reach a scientific truth solely by this approach. In order to facilitate scientific discoveries from single-cell gene expression data, we need a <em>quantitative</em> methodology.</p>
<p>This is why we developed Treefit - the first software for performing <em>quantitative</em> trajectory inference. Treefit can be used in conjunction with existing popular software packages such as <a href="https://satijalab.org/seurat/" class="external-link">Seurat</a> and <a href="https://dynverse.org/" class="external-link">dynverse</a>. In this tutorial, we demonstrate how to use Treefit together with Seurat by analyzing the single-cell RNA-seq dataset in <a href="https://www.nature.com/articles/nbt.2859" class="external-link">Trapnell <i>el al.</i>, 2014, Nature Biotechnology</a>.</p>
</div>
<div class="section level2">
<h2 id="trajectory-inference-using-treefit-and-seurat">2. Trajectory inference using Treefit and Seurat<a class="anchor" aria-label="anchor" href="#trajectory-inference-using-treefit-and-seurat"></a>
</h2>
<p>As the Treefit package is meant to be used with gene expression data (either raw counts or normalized values) that have been already quality controlled, it is a good idea to use Seurat or other popular toolkits designed for the quality control of single-cell gene expression data (<i>e.g.</i>, filtering cells and normalizing the data) before the Treefit analysis.</p>
<div class="section level3">
<h3 id="loading-the-dataset">2.1. Loading the dataset<a class="anchor" aria-label="anchor" href="#loading-the-dataset"></a>
</h3>
<p>We analyze the dataset provided in <a href="https://www.nature.com/articles/nbt.2859" class="external-link">Trapnell <i>el al.</i>, 2014, Nature Biotechnology</a>. This dataset was originally acquired for the purpose of studying the process of differentiation of human myoblasts that can be reasonably explained by the linear model (<i>i.e.</i>, non-branching, chain-like structure). However, as the authors discuss in detail in the original article, their trajectory inference software based on a minimum spanning tree (MST) helped detect contamination with another cell lineage which was not related to myogenesis and so revealed that the bifurcation tree model (<i>i.e.</i>, Y-shaped tree structure) fits the data better than the linear model.</p>
<p>To use this dataset, we download and read the dataset provided in dynverse as follows (see <a href="https://zenodo.org/record/1443566" class="external-link uri">https://zenodo.org/record/1443566</a> for details).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trapnell.path</span> <span class="op">&lt;-</span> <span class="st">"myoblast-differentiation_trapnell.rds"</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">trapnell.path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://zenodo.org/record/1443566/files/real/gold/myoblast-differentiation_trapnell.rds?download=1"</span>,
                <span class="va">trapnell.path</span>,
                mode<span class="op">=</span><span class="st">"wb"</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">trapnell.dynverse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">trapnell.path</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="preprocessing-with-seurat">2.2. Preprocessing with Seurat<a class="anchor" aria-label="anchor" href="#preprocessing-with-seurat"></a>
</h3>
<p>From now, we demonstrate how to preprocess the above data with Seurat. In order to handle data in Seurat, we first need to create a <code>Seurat</code> object. When creating a <code>Seurat</code> object from dynverse data, we must be aware that the roles of the rows and columns are reversed and therefore we need to transpose the matrix. In dynverse data, the rows and columns correspond to cells and genes, respectively, and vice versa in <code>Seurat</code> objects.</p>
<p>Here we assume that we wish to perform the following preprocessing.</p>
<ul>
<li>Excluding each gene that is only expressed in less than 3 cells</li>
<li>Removing each cell that only expresses less than 200 genes</li>
</ul>
<p>This can be done by setting the designated parameters as follows.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trapnell</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">trapnell.dynverse</span><span class="op">$</span><span class="va">count</span><span class="op">)</span>,
                                       min.cells<span class="op">=</span><span class="fl">3</span>,
                                       min.features<span class="op">=</span><span class="fl">200</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Feature names cannot have underscores ('_'), replacing with dashes</span>
<span class="co">## ('-')</span></code></pre>
<p>The “Feature names cannot have underscores …” warning message can be ignored. It’s caused by difference between dynverse and Seurat. Dynverse uses underscores for gene names but Seurat doesn’t allow it. Seurat replaces underscores with dashes automatically with the warning message.</p>
</div>
<div class="section level3">
<h3 id="visualization-with-seurat">2.3. Visualization with Seurat<a class="anchor" aria-label="anchor" href="#visualization-with-seurat"></a>
</h3>
<p>Although pictures may not always describe reality, we usually visualize data before starting detailed analysis. Here we create a scatter plot by principal component analysis (PCA). Below are a few lines of code to do this and the result is shown in Figure 1.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trapnell</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">trapnell</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">trapnell</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">trapnell</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">trapnell</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">trapnell</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Embeddings.html" class="external-link">Embeddings</a></span><span class="op">(</span><span class="va">trapnell</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="working-with-seurat_files/figure-html/visualization-1.png" alt="Figure 1. The result of PCA for the myoblast dataset" width="700"><p class="caption">
Figure 1. The result of PCA for the myoblast dataset
</p>
</div>
<p>It is hard to reach the truth from the PCA result shown in Figure 1. Unlike the tree-like toy data analyzed in the previous tutorial (<code><a href="../articles/treefit.html">vignette("treefit")</a></code>), we cannot easily recognize the underlying tree structure for various reasons (<i>e.g.</i>, high level of noise, contamination with different types of cells, and the inaccuracy of two-dimensional visualization of high dimensional data).</p>
<p>Figure 2b in <a href="https://www.nature.com/articles/nbt.2859" class="external-link">Trapnell <i>el al.</i>, 2014, Nature Biotechnology</a> shows that this dataset contains three distinct types of cells (<i>i.e.</i>, proliferating cells, differentiating myoblasts, and contaminating interstitial mesenchymal cells), which led the authors to adopt a Y-shaped tree model (<i>i.e.</i>, a star tree with three arms) to explain the data. However, it is difficult to quantify the reliability of this tree model and to confidently predict that the best-fit tree consists of three principal paths.</p>
</div>
</div>
<div class="section level2">
<h2 id="quantitative-trajectory-inference-with-treefit">3. Quantitative trajectory inference with Treefit<a class="anchor" aria-label="anchor" href="#quantitative-trajectory-inference-with-treefit"></a>
</h2>
<div class="section level3">
<h3 id="performing-the-treefit-analysis">3.1. Performing the Treefit analysis<a class="anchor" aria-label="anchor" href="#performing-the-treefit-analysis"></a>
</h3>
<p>Let us use Treefit in order to estimate the tree-likeness of the preprocessed myoblast data (which contain 290 cells) and to predict the number of principal paths in the best-fit tree. The workflow is the same as the toy data analysis described in the previous tutorial (<code><a href="../articles/treefit.html">vignette("treefit")</a></code>).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trapnell.fit</span> <span class="op">&lt;-</span> <span class="fu">treefit</span><span class="fu">::</span><span class="fu"><a href="../reference/treefit.html">treefit</a></span><span class="op">(</span><span class="va">trapnell</span><span class="op">)</span>
<span class="va">trapnell.fit</span></code></pre></div>
<pre><code><span class="co">## $name</span>
<span class="co">## [1] "trapnell"</span>
<span class="co">## </span>
<span class="co">## $max_cca_distance</span>
<span class="co">##     p      mean standard_deviation</span>
<span class="co">## 1   1 0.6303459         0.12773201</span>
<span class="co">## 2   2 0.4017539         0.07430735</span>
<span class="co">## 3   3 0.3255355         0.05479342</span>
<span class="co">## 4   4 0.2699334         0.04228203</span>
<span class="co">## 5   5 0.2500198         0.04335692</span>
<span class="co">## 6   6 0.2348225         0.03919659</span>
<span class="co">## 7   7 0.2269746         0.03631043</span>
<span class="co">## 8   8 0.2185274         0.03309986</span>
<span class="co">## 9   9 0.2096228         0.03344765</span>
<span class="co">## 10 10 0.2022596         0.03228294</span>
<span class="co">## 11 11 0.1928535         0.03540076</span>
<span class="co">## 12 12 0.1878315         0.03622790</span>
<span class="co">## 13 13 0.1811885         0.03879294</span>
<span class="co">## 14 14 0.1725552         0.03903983</span>
<span class="co">## 15 15 0.1501188         0.03806641</span>
<span class="co">## 16 16 0.1423743         0.03696181</span>
<span class="co">## 17 17 0.1332141         0.03425474</span>
<span class="co">## 18 18 0.1301240         0.03567304</span>
<span class="co">## 19 19 0.1261766         0.03589544</span>
<span class="co">## 20 20 0.1229404         0.03627929</span>
<span class="co">## </span>
<span class="co">## $rms_cca_distance</span>
<span class="co">##     p      mean standard_deviation</span>
<span class="co">## 1   1 0.6303459         0.12773201</span>
<span class="co">## 2   2 0.5199144         0.06498940</span>
<span class="co">## 3   3 0.6170262         0.06686113</span>
<span class="co">## 4   4 0.6407349         0.06804858</span>
<span class="co">## 5   5 0.6547144         0.05073100</span>
<span class="co">## 6   6 0.5958460         0.06191394</span>
<span class="co">## 7   7 0.5788654         0.05568816</span>
<span class="co">## 8   8 0.6097528         0.04763631</span>
<span class="co">## 9   9 0.6087060         0.05660548</span>
<span class="co">## 10 10 0.6030251         0.04738148</span>
<span class="co">## 11 11 0.6011495         0.04392654</span>
<span class="co">## 12 12 0.6174355         0.03846905</span>
<span class="co">## 13 13 0.6230418         0.03887509</span>
<span class="co">## 14 14 0.6230970         0.03439415</span>
<span class="co">## 15 15 0.6267405         0.03635980</span>
<span class="co">## 16 16 0.6248394         0.03573201</span>
<span class="co">## 17 17 0.6311507         0.02858439</span>
<span class="co">## 18 18 0.6413880         0.02845259</span>
<span class="co">## 19 19 0.6425895         0.03337487</span>
<span class="co">## 20 20 0.6486755         0.02962193</span>
<span class="co">## </span>
<span class="co">## $n_principal_paths_candidates</span>
<span class="co">## [1]  3  8 12 17</span>
<span class="co">## </span>
<span class="co">## attr(,"class")</span>
<span class="co">## [1] "treefit"</span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">trapnell.fit</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="working-with-seurat_files/figure-html/plot-1.png" alt="Figure 2. Summary of the Treefit analysis results" width="700"><p class="caption">
Figure 2. Summary of the Treefit analysis results
</p>
</div>
</div>
<div class="section level3">
<h3 id="interpreting-the-results">3.2. Interpreting the results<a class="anchor" aria-label="anchor" href="#interpreting-the-results"></a>
</h3>
<p>As explained in the previous tutorial (<code><a href="../articles/treefit.html">vignette("treefit")</a></code>), the two Grassmann distances <code>$max_cca_distance</code> and <code>$rms_cca_distance</code> play an essential role in the Treefit analysis. The goal of the first analysis using <code>$max_cca_distance</code> is to measure the goodness-of-fit between data and data-derived tree trajectories, and this helps biologists make a mathematical and statistical evidence-based quantitative discussion. The other analysis with <code>$rms_cca_distance</code> aims to predict the number of principal paths in the best-fit tree trajectory, and the result of this analysis can be helpful to discover novel cell types or make sure whether or not there are contaminating cell types.</p>
<p>If we compare the mean values of <code>$max_cca_distance</code> shown in the left panel of Figure 2 with the previous results (<code><a href="../articles/treefit.html">vignette("treefit")</a></code>), we can see that the tree-likeness of the myoblast dataset is more like the noisy tree-like data with the <code>fatness</code> <code>0.8</code> rather than the tree-like data with the <code>fatness</code> <code>0.1</code>. This estimate is reasonable in light of the fact that the myoblast dataset in <a href="https://www.nature.com/articles/nbt.2859" class="external-link">Trapnell <i>el al.</i>, 2014, Nature Biotechnology</a> contains some contaminating cell types that are not related to myogenesis.</p>
<p>Turning to the right panel of Figure 2, we see that <code>$rms_cca_distance</code> attains the local minimum at <i>p</i>=2. Therefore, we can deduce that the best-fit tree trajectory for the myoblast data consists of <i>p</i>+1=3 principal paths. <code>$n_principal_paths_candidates[1]</code> also tells us the number principal paths is 3. Recalling the discussion in <a href="https://www.nature.com/articles/nbt.2859" class="external-link">Trapnell <i>el al.</i>, 2014, Nature Biotechnology</a> (<i>i.e.</i>, the cells in the dataset can be classified into the following three groups: proliferating cells; differentiating myoblasts; and contaminating interstitial mesenchymal cells), we thus see that Treefit has correctly predicted the number of distinct principal paths forming the best-fit tree trajectory for this dataset.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Momoko Hayamizu, Kouhei Sutou, Ryohei Suzuki, Hiromi Ishii.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
